### PR TITLE
Code Cleanup and Fluent Interface

### PR DESCRIPTION
**Pull request by MovGP0**

### CODE REVIEW

File: test/shience.test/ScienceTests.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            }
+
+            [Fact]
+            public void ComparerFuncReturnsCorrectFalseResult()
+            {
+                var science = Shience.New<bool>("ComparerFuncReturnsCorrectFalseResult");
+
+                var result = science.Test(control: () => true,
+                    candidate: () => false,
+                    comparer: (a, b) => a == b).Execute();
```
Comment (davezych) [inline test/shience.test/ScienceTests.cs]: There's no overload of `Test` with a comparer now. This needs to use `WithComparer`.

File: test/shience.test/ScienceTests.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            }
+            
+            [Fact]
+            public async Task TestsAreRunInParallel()
+            {
+                var science = Shience.New<bool>("TestsAreRunInParallel");
+
+                var output = string.Empty;
+
+                var result = await science.TestAsync(
```
Comment (davezych) [inline test/shience.test/ScienceTests.cs]: There's no `TestAsync` method, this should use `Test.ExecuteAsync`.

**Comment by davezych** [comment]
I noted a few build errors that need fixing, otherwise looks good.

**Comment by davezych** [comment]
Oh, also, the [Readme](https://github.com/davezych/shience/blob/master/readme.md) needs to be updated to reflect the fluent interface.

**Comment by davezych** [comment]
One more thought: should the control and candidate be moved out to their own methods? i.e. instead of:

    science.Test("test-name", control: () => { /*whatever*/ }, candidate: () => { /*Whatever */ })

something like:

    science.Test("test-name")
                .Control(() => { /*whatever*/ })
                .Candidate(() => { /*Whatever */ }
                .Execute()

Methods could be named `Use`/`Try` as well to stay consistent with the wording that Scientist uses.

**Comment by MovGP0** [comment]
sorry for the build errors. could not fix them because the reference in the test project was not working properly.

**Comment by MovGP0** [comment]
the build errors are fixed and #16 is solved too

**Comment by davezych** [comment]
When I pull this down and restore packages the test lib is not able to find the Shience package. If I move the dependency into the `dnxcore50` framework I'm able to restore. Not sure why that matters, seeing as we're only targeting `dnxcore50` anyway. You didn't run into that?

Everything else looks good, once we get this figured out I'll merge it.

**Comment by MovGP0** [comment]
strange. on my machine both versions are working. not sure what the difference is.

**Comment by davezych** [comment]
I changed the casing from `Shience` to `shience` and it restored. I'm going to chalk it up to a bug in tooling.

This all looks good, merging... :shipit:

**Pull request merged**